src/workerd/api/wpt/url-test.js

```diff
@@ -0,0 +1,42 @@
+import {
+  strictEqual,
+} from 'assert';
+
+globalThis.fetch = async(url) => {
+  const { default: data } = await import(url);
+  return {
+    async json() { return data; },
+  };
+};
+
+globalThis.promise_test = (callback) => {
+  callback();
+};
+
+globalThis.assert_equals = (a, b, c) => {
+  strictEqual(a,b,c);
+};
+
+globalThis.test = (callback, message) => {
+  try {
+    callback();
+  } catch (err) {
+    const aerr = new AggregateError([err], message);
+    globalThis.errors.push(aerr);
+  }
+}
+
+globalThis.errors = [];
+
+export const test = {
+  async test(_, env) {
+    const foo = await import('url-origin.any.js');
```

Comment:
The tests end up running when the module is evaluated. We have to set the global test harness methods before the dynamic import, then check the `globalThis.errors` when we're done. The `globalThis.errors` is only specific to this test file, it is not actually part of the WPT test harness. We have to do this because the tests themselves may not actually propagate the error (in the case of url-origin.any.js we end up with a swallowed rejected promise) if we don't have this.

Reply:
We also need a way to ignore certain errors. Possibly from a `module_name.json` file with a JSON schema that conforms to our needs

Reply:
Yeah, I'm thinking that can be a separate json module that is imported and used by the harness.